### PR TITLE
Multi-Factor Authentication Using Redis Cache and OTP

### PR DESCRIPTION
**Pull request by divofred**
#### Please update the following checklist as applicable. Otherwise, the maintainers may close your pull request.

- [x] You created a branch from `develop` branch.
- [x] You PR is raised on `develop` branch and not on`main`.
- [x} Blog follows [Blog Guidelines](https://github.com/LoginRadius/engineering-portal/blob/main/GUIDELINES.md).
- [x] You have read the [Contribution Guidelines](https://github.com/LoginRadius/engineering-portal/blob/main/CONTRIBUTING.md) before creating this PR.
- [x] Checked for any duplicate blog entry.
- [x] Checked for all grammatical errors in the blog.
- [x] You have not published the blog content elsewhere on the internet on platforms like Medium, Dev.to, etc.
- [x] The content is not copy-pasted from elsewhere on the internet.

### CODE REVIEW

**Comment by raghunath-r-a** [comment]
Hi @divofred We did PHASE I review and moving your PR to PHASE II review.

If you want to add the sample code for easier reference, please add sample code in the following repo by creating a separate PR: https://github.com/LoginRadius/engineering-blog-samples/ 

Thanks :)
@raghunath-r-a

**Comment by divofred** [comment]
Thanks @raghunath-r-a,
I've created a PR to the GitHub Repository.

Regards,
Fredrick Emmanuel.

File: content/engineering/guest-post/multi-factor-authentication-using-redis-cache-and-otp/index.md [hunk 1]
```diff
@@ -0,0 +1,628 @@
...
+1. In the terminal, install the `nanoid` module. This module will create a random set of characters as the OTP. Run the following command in your terminal:
+    
+    ```bash
+    npm i nanoid
+    ```
+    
+2. After installing the `nanoid` module, import it into `server.js` and use it to generate 6 random digits.
+    
+    ```js
+    const nanoid = require("nanoid");//Importing the naonid module
```
Comment (VershaGupta) [inline content/engineering/guest-post/multi-factor-authentication-using-redis-cache-and-otp/index.md:177]: @divofred  Got the following error after adding the `nanoid` module
image

Comment (divofred) [reply content/engineering/guest-post/multi-factor-authentication-using-redis-cache-and-otp/index.md:177]: @VershaGupta I'm sorry you got that error. I will fix it ASAP.

Comment (divofred) [reply content/engineering/guest-post/multi-factor-authentication-using-redis-cache-and-otp/index.md:177]: @VershaGupta I've fixed the error. The latest version of nanoid doesn't support the use of `require` to import the module.
I specified the version of nanoid compatible with NodeJS.

**Review by VershaGupta — APPROVED** [review]
PR  looks good

**Pull request merged**